marshal/consumer.go

```diff
@@ -458,76 +485,59 @@ func (c *Consumer) claimTopics() {
 	}
 }
 
-// updatesTopicClaims if changed. It should be noted that it expects c.lock.RLock
-// to be already acquired
-func (c *Consumer) safeUpdateTopicClaims(latestClaims map[string]bool, force bool) {
```

Comment:
it's important that we don't send the same notification multiple times. I don't believe you're breaking this behavior, but depending on how you handle losing topic claims per my comment above, this might be affected.